c20_game_of_life/game_of_life.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+type universe [][]bool
+
+// Creates a new instance of a universe - every cell is initialized to false
+func newUniverse(height int, width int) universe {
+	h := make([][]bool, height)
+	for w, _ := range h {
+		h[w] = make([]bool, width)
+	}
+	return h
+}
+
+// prints out a universe's 2d array, using 1 for true and 0 for false
+func (u universe) showUniverse() {
+	c0 := "0"
+	c1 := "1"
+	for _, h := range u {
+		for _, w := range h {
+			if w {
+				fmt.Printf(" %v", c1)	
+			} else {
+				fmt.Printf(" %v", c0)
+			}
+		}
+		fmt.Printf("\n")
+	}
+}
+
+// calculates the percentage of cells that are currently alive
+func (u universe) calcPopPercent() float64 {
+	var totalCells int
+	var liveCells int
+	for _, h := range u	{
+		totalCells += len(h)
+		for w, _ := range h {
+			if h[w] == true {
+				liveCells++
+			}
+		}
+	}
+	return float64(liveCells) / float64(totalCells) * 100
+}
+
+// prints a universe's state and percentage of living cells
+func (u universe) printUniverseInfo() {
+	u.showUniverse()
+	fmt.Printf("living cell percentage: %v%%\n", u.calcPopPercent())
+}
+
+// seeds a universe, with a 25% chance for each cell to become alive
+func (u universe) seedUniverse() {
+	rand.Seed(time.Now().UnixNano())
+	for _, h := range u	{
+		for w, _ := range h {
+			h[w] = rand.Intn(4) == 3
+		}
+	}
+}
+
+// If x or y are out of range, it loops the coordinate to the other side
+func (u universe) treatCoord(y int, x int) (int, int) {
```

Comment:
Finally, you might want to change calling the variables `x` and `y` because it looks like they are now `h` and `w`, respectively, which doesn't make sense in a Cartesian plane.

Reply:
I had a hard time figuring out a good name for this at the time, so I called it treatCoord. The idea is that out of bounds coordinates would be "treated", and return a "treated" set of coordinates. It's not the best naming convention. Perhaps "loopCoords" or something?
Having the function accept a number and a length is a good idea. I think at this time, I was both in a work-flow, and also infatuated with the ability to return multiple variables.
I'm not necessarily thinking of this in a mathematical context - I just use the terms x/width and y/height to keep track of which should be paired, or which variables corollate with the other. If it makes it more readable, then I'll consider renaming them.